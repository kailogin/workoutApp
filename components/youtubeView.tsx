import React from "react";
import { View, Platform, StyleSheet } from "react-native";
import { WebView } from "react-native-webview";

interface YouTubeViewProps {
  youtubeId: string;
}
export const YouTubeView = ({ youtubeId }: YouTubeViewProps) => {
  return (
    <View style={{ height: 300 }}>
      <WebView
        style={styles.WebViewContainer}
        javaScriptEnabled={true}
        domStorageEnabled={true}
        source={{ uri: `https://www.youtube.com/embed/${youtubeId}` }}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  WebViewContainer: {
    margin: 20,
  },
});
