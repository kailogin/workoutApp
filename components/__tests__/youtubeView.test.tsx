import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import { YouTubeView } from "../youtubeView";

test(`BaseText snapshot`, () => {
  const snap = renderer.create(<YouTubeView youtubeId="testMe" />).toJSON();

  expect(snap).toMatchSnapshot();
});
