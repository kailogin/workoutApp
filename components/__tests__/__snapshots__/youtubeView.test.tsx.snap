// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseText snapshot 1`] = `
<View
  style={
    Object {
      "height": 300,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      domStorageEnabled={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "uri": "https://www.youtube.com/embed/testMe",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "margin": 20,
          },
        ]
      }
      useSharedProcessPool={true}
    />
  </View>
</View>
`;
